executor/resources/sys_nolinux.go

```diff
@@ -0,0 +1,13 @@
+//go:build !linux
+
+package resources
+
+import (
+	"time"
+
+	"github.com/moby/buildkit/executor/resources/types"
+)
+
+func newSysSampler() (*Sampler[*types.SysSample], error) {
+	return NewSampler(2*time.Second, 20, func(_ time.Time) (*types.SysSample, error) { return nil, nil }), nil
+}
```

Comment:
TBH; wasn't sure if `return nil, nil` would work, but looks like it's happy

Also was hoping to make this function return `nil`, but not sure if that would work, as it's called unconditionally; https://github.com/moby/buildkit/blob/d9a6afdf089a7c4b97cac704a60ad70c21086f12/solver/llbsolver/solver.go#L419-L420

I was initially thinking along the lines of

```go
var usage some.Type
if s.sysSampler != nil {
    usage := s.sysSampler.Record()
    defer usage.Close(false)
}

```

But honestly, generics are not yet "my thing", and I got confused what to use for `some.Type` (`.Record()` has a `func (s *Sampler[T]) Record() *Sub[T]` signature)

Reply:
The type should just be `*Sub[*types.SysSample]`.